tasks/insights-client.yml

```diff
@@ -36,6 +36,59 @@
     insertafter: "#auto_update"
     line: auto_update={{ rhc_insights.autoupdate | d(true) | bool }}
 
+- name: Check ansible host in insights-client config
+  when:
+    - rhc_insights.ansible_host is defined
+    - not rhc_insights.ansible_host is none
+    - rhc_insights.ansible_host != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^ansible_host="
+    state: "{{ 'present' if rhc_insights.ansible_host != __rhc_state_absent else 'absent' }}"
+    line: "{{ ('ansible_host='+rhc_insights.ansible_host) if rhc_insights.ansible_host != __rhc_state_absent }}"
+  check_mode: true
+  register: __insights_ansible_host_exists
+
+- name: Configure ansible host
+  when:
+    - rhc_insights.ansible_host is defined
+    - not rhc_insights.ansible_host is none
+    - rhc_insights.ansible_host != omit
+    - __insights_ansible_host_exists.changed
+  block:
+    - name: Update ansible host in insights-client config
+      lineinfile:
+        path: "{{ __rhc_insights_conf }}"
+        regexp: "^ansible_host"
+        insertafter: "#ansible_host"
+        line: ansible_host={{ rhc_insights.ansible_host }}
+    - name: Update ansible host in inventory
+      shell: insights-client --ansible-host={{ rhc_insights.ansible_host }} & wait
+      when: 
+        - rhc_insights.ansible_host != __rhc_state_absent
+        - >-
+          "This host is registered" in __rhc_insights_status.stdout
+          or "Registered" in __rhc_insights_status.stdout
+
+- name: Remove ansible host
+  when:
+    - rhc_insights.ansible_host is defined
+    - rhc_insights.ansible_host == __rhc_state_absent
+    - __insights_ansible_host_exists.changed
```

Comment:
this condition can be dropped altogether: if we need to remove the `ansible_host`, then there is no need to check that it exists beforehand, `lineinfile` will do that

Reply:
I ran into some unexpected behavior with this suggestion.  Removing  the entire condition altogether  breaks the setting/updating of the `ansible_host` value in HBI since it results in the 2nd task in the block to always execute.

Did you mean to ask me to only remove this?
```
    - __insights_ansible_host_exists.changed
```
Removing only this condition works as expected with the other suggestions you've made.